### DISCUSSION TITLE
Zustand 저장소의 데이터(액세스 토큰, 유저 개인정보) 관리 방법에 대해 의견이 필요합니다!

### DISCUSSION THREAD

**Discussion by Yoonyesol** [GU-99/grow-up-fe#92] [질문과 답변]
**문제상황**
새로고침 시 zustand 저장소의 데이터가 초기화되는 문제를 해결하기 위해, `persist`라는 미들웨어를 사용하여 데이터를 브라우저 저장소에 저장하였습니다. 이로 인해 액세스 토큰이나 유저의 개인정보와 같은 주요 정보가 local storage에 저장되어 보안적인 이슈가 있는 상황입니다.

**질문**
우선 유저 정보의 경우 zustand 대신, 리액트 쿼리의 캐싱기능을 이용하면 다른 페이지에서도 통신을 통해 가져온 유저 정보를 사용할 수 있다고 해서 적용해 볼 생각입니다. (https://careerly.co.kr/qnas/6286)
액세스 토큰 같은 경우는 로컬 스토리지 암호화나 쿠키에 저장 후 보안 속성 적용 등... 여러가지를 생각해보고는 있는데 아직 보안적으로 괜찮은 방법을 찾지 못했습니다😭
액세스 토큰을 어떤 방식으로 저장해야 좋을지, 다른 분들의 의견을 들어보고 싶습니다!
또 `persist` 적용 없이 새로고침 시에도 zustand의 데이터 내용을 저장할 수 있는 방법을 알고 계시다면, 방법을 공유해주시면 정말 감사하겠습니다🙇‍♂️

**Comment by Seok93** [comment]
zustand persist를 사용하면 session storage나 local storage에 저장되는데, 부분 영속화를 할 수 있는 옵션이 있는걸로 알고 있습니다.  [zustand: partialize](https://zustand.docs.pmnd.rs/integrations/persisting-store-data#partialize)를 한 번 참고해보면 좋을 것 같아요.  그래서 필요한 정보들은 영속화를 하되, access token은 제외하면 어느 정도 보완할 수 있는 부분이라고 생각합니다. 그러면 여전히 남는 문제는 새로고침을 하면 access token이 날아가는 부분인데, 이는 refresh token이 있어서 재발급이 진행되니 사용할 수 있는 방법이라고 생각은 하지만, 잦은 새로고침으로 매번 서버에 요청하는게 리소스를 낭비하는 모양새가 될 수 있을 것 같아요. 그런 의미에서 말씀하신 리액트 쿼리 캐싱 기능으로 관리하는 것도 괜찮을 것 같아요. 예전에 봤던 강의에서도 비슷하게 관리하던 코드를 본 것 같습니다.

**Reply by Yoonyesol** [reply]
우석님 답변 감사합니다! zustand 영속화는 처음 들어보는 개념인데 리액트 쿼리 캐싱과 함께 공부해 봐야 할 것 같습니다.관련해서 참고할 수 있는 강의가 있는지도 서치해봐야겠네요👍👍 두 가지 개념 사용해서 데이터 유지 구현해보고, 팀원분들 의견을 들어야 할 사항이 생긴다면 다시 공유 드리겠습니다

**Reply by Seok93** [reply]
영속화는  `persist` 미들웨어를 통해 새로고침시 사라지는 데이터를 지속적으로 관리할 수 있는 성질을 부여하는 것을 말해요. 예솔님이 사용하시는 persist가 바로 그 개념입니다😉

※ [영속화(永續化)](https://dict.naver.com/dict.search?dicQuery=%E6%B0%B8%E7%BA%8C%E5%8C%96&query=%E6%B0%B8%E7%BA%8C%E5%8C%96&target=dic&ie=utf8&query_utf&isOnlyViewEE): 어떤 일이 중도에 끊기거나 바뀌지 않고 지속됨

**Reply by Yoonyesol** [reply]
앗 부분 영속화...를 말한다는 걸 제가 영속화라고 말씀드렸네요😆ㅋㅋㅋㅋ 감사합니다! partialize 개념 공부 해보겠습니다!!😊😎
